service/src/main/java/io/camunda/service/RoleServices.java

```diff
@@ -52,24 +53,25 @@ public RoleServices withAuthentication(final Authentication authentication) {
         brokerClient, securityContextProvider, roleSearchClient, authentication);
   }
 
-  public CompletableFuture<RoleRecord> createRole(final RoleDTO request) {
-    return sendBrokerRequest(new BrokerRoleCreateRequest().setName(request.name()));
+  public CompletableFuture<RoleRecord> createRole(final String name) {
+    return sendBrokerRequest(new BrokerRoleCreateRequest().setName(name));
   }
 
-  public CompletableFuture<RoleRecord> updateRole(final RoleDTO request) {
-    return sendBrokerRequest(
-        new BrokerRoleUpdateRequest(request.roleKey()).setName(request.name()));
+  public CompletableFuture<RoleRecord> updateRole(final long roleKey, final String name) {
+    return sendBrokerRequest(new BrokerRoleUpdateRequest(roleKey).setName(name));
   }
 
-  public RoleEntity getByRoleKey(final Long roleKey) {
-    final SearchQueryResult<RoleEntity> result =
-        search(SearchQueryBuilders.roleSearchQuery().filter(f -> f.roleKey(roleKey)).build());
-    if (result.total() < 1) {
-      throw new NotFoundException(String.format("Role with roleKey %d not found", roleKey));
-    } else {
-      return result.items().stream().findFirst().orElseThrow();
-    }
+  public RoleEntity getRole(final Long roleKey) {
+    return findRole(roleKey)
+        .orElseThrow(
+            () -> new NotFoundException("Role with roleKey %d not found".formatted(roleKey)));
   }
 
-  public record RoleDTO(long roleKey, String name, Set<Long> assignedMemberKeys) {}
+  @VisibleForTesting
+  public Optional<RoleEntity> findRole(final Long roleKey) {
+    return search(SearchQueryBuilders.roleSearchQuery().filter(f -> f.roleKey(roleKey)).build())
+        .items()
+        .stream()
+        .findFirst();
```

Comment:
❓ Should we protect against scenarios where we find multiple entries with this key? Be it for issues on the data layer or refactorings that mess this up later (e.g. renaming fields in the data layer and forgetting to adjust this in the search layer). The `UserTaskServices` do something like this. In the end, the contract of this method should be to find the one role with this key and to indicate when this wasn't possible (i.e. not found or too many found).

Reply:
I agree, there's a trade-off though: with a similar argument, we could check that the returned results actually have the requested key (this is not completely unthinkable given that we're querying ES). But the additional code is unlikely to pay off and hurts readability. So we need to draw the line somewhere. 

I don't want to turn this into an argument though, if you prefer to check for multiple results that works for me.

Reply:
Get your point and agree, we have to draw the line somewhere 👍
I am fine with keeping things as they are here specifically 👍